### DISCUSSION TITLE
Implicit null

### DISCUSSION THREAD

**Discussion by ProtocolLive** [rectorphp/rector#8909] [Q&A] [answered]
The Rector are not detecting the implicit null deprecation
```php
return RectorConfig::configure()
->withPhpSets(php84: true);
```
```php
function foo(int$bar = null){}
```
```
Deprecated: foo(): Implicitly marking parameter $bar as nullable is deprecated, the explicit nullable type must be used instead in php shell code on line 1
```

**Comment by samsonasik** [comment, marked as answer]
If you don't have runtime php 8.4 or have but have composer.json not define `"php": ^8.4`, you can enforce it via : `->withPhpVersion(PhpVersion::PHP_84)`, see

https://getrector.com/demo/75fcfb81-0ff1-4194-a5e0-603813d3f2d5

for note: upgrading code php should be step by step, see https://github.com/rectorphp/rector/discussions/8823#discussioncomment-10599155 for reasoning.

**Comment by ProtocolLive** [comment]
Works, thanks
